res/controllers/Denon-Prime-4-scripts.js

```diff
@@ -321,16 +345,50 @@ Prime4.Deck = function(deckNumbers, midiChannel) {
             this.type = components.Button.prototype.types.push;
         }
     });
-    // Hotcue Pads
-    this.hotcuePad = [];
-    for (let i = 1; i <= 8; i++) {
-        this.hotcuePad[i] = new components.HotcueButton({
-            number: i,
-            midi: [0x90 + midiChannel, 0x0E + i],
-            colorMapper: Prime4ColorMapper,
-            off: Prime4.rgbCode.whiteDark,
-        });
+
+    // Performance Pads
+    // TODO: Figure out how to work with different pad modes
+    this.performancePad = [];
+
+    let padMode = "HOT CUE"; // Set to Hotcue mode by default for now, while I work on handling other pad modes.
+
+    /********* BEGIN REVIEW REQUEST **********/
+    // These are my attempt at handling the pad modes being switched. Unsuccessful so far.
+    // I can call the functions OK, but the padMode variable changing doesn't seem
+    // to update the pads themselves. Not sure how to handle this with Components
+    // yet.
```

Comment:
So I struggled with this myself while working on my NS6II mapping. I tried a bunch of different implementations but in the end the best is the one thats also employed in the Roland DJ505 mapping (which is excellent btw, I highly recommend you use that as a reference). 
To summarize the approach: You create a new componentcontainer that contains the buttons that "dial" the performance modes (hotcue, loop, roll, slicer). Each of your decks gets one instance of that. Then you create a custom componentcontainer for each of your pad modes (and the parameter left/right buttons are usually also part of that container). Then in the performance-mode-componentcontainer, you instantiate each of your containers once but only ever connect a single container at a time. Its a little bit convoluted but thats basically the approach taken by the DJ505 mapping (see `DJ505.PadSection`). Also my Numark NS6II mapping does pretty much the same, so you can take a look at that too (though I'm using a little more advanced javascript using the lodash library, so it's likely to be a little too complicated for you). Look for `NS6II.PadModeContainers` (especially `.ModeSelector`) in https://github.com/mixxxdj/mixxx/pull/11075/files

For the future, we might add another componentsJS class to simplify this process.